### DISCUSSION TITLE
Starter for "db app" without user, mailer, worker, etc

### DISCUSSION THREAD

**Discussion by felipesere** [loco-rs/loco#125] [Q&A] [answered]
👋 I want to migrate a small go app to `loco-rs`. It already has a database with a bunch of data though.
I configured my local db and ran `cargo loco db entities` but that now complains that all the user entities are gone.
Which makes total sense! Those entities never existed in my DB.
So I wonder if there is a case to be made for a simpler starter that is just wired up for a DB?
Basically something like this:
```bash
$ loco new foody --db --no-mailer --no-user-auth --no-worker
```

**Comment by kaplanelad** [comment]
Hey @felipesere, great call on migrating to Loco! :)

we are trying to avoid creating starters per specific usecas (there is a lot of combination) to make a choice easier for the user and also for less maintenance. We build `Loco` modular which always you to turn off some component that you are not needs in a simply way.

**Auth**: The Auth is a  feature flag that is enabled by default. If you don't need it, you can turn it off [here](https://github.com/loco-rs/loco/blob/master/Cargo.toml#L15).

**Mailer**: We currently consider the Mailer as part of the SaaS tool. If you don't require it, you can remove the loader in `lib.rs`. This action will make the mailer unavailable. Feel free to keep the folder for future use or remove it altogether.

**worker** - By default, the worker is not loaded when you run loco. run `cargo loco start --help` and see that the worker not loading by default. If this aligns with your preferences, great! You can also remove the worker registration [here](https://github.com/loco-rs/loco/blob/master/starters/saas/src/app.rs#L36), and your app won't have workers at all.

**Reply by felipesere** [reply]
Yeah makes total sense to keep a limit on the branches. Plus I can just delete code where I need to.

I realised why I was getting compile errors about user and notes: I think the mod.rs file gets overwritten when running `cargo loco db entities`? Which makes sense if there is to be a strict 1:1 mapping. 
I had to add the modules back into `models/_entities/mod.rs`.
Will try to add the user & notes tables as I can totally use that for my project.

I'll keep posting here with my experience as I go through it 😅

**Reply by kaplanelad** [reply, marked as answer]
@felipesere Happy to hear! We are here to assist with any further questions you may have. Feel free to share new ideas or provide feedback that could contribute to our continuous improvement.

Please do keep us updated on your progress, and let us know when the service is live in production.:)
